contracts/MetawoodAuction.sol

```diff
@@ -0,0 +1,330 @@
+        _;
+    }
+
+    modifier ensureValidAuction(uint256 _auctionId) {
+        Auction memory auction = _auctions[_auctionId];
+        require(auction.creator != address(0), "Invalid Auction");
+        require(msg.sender != address(0), "Invalid caller address");
+        _;
+    }
+
+    modifier ensureActiveAuction(uint256 _auctionId) {
+        Auction memory auction = _auctions[_auctionId];
+        require(auction.status == AuctionState.ACTIVE, "Auction not active");
+        _;
+    }
+
+    modifier ensureAuctionCreator(uint256 _auctionId) {
+        Auction memory auction = _auctions[_auctionId];
+        require(auction.creator == msg.sender, "Not the auction creator");
+        _;
+    }
+
+    // Transactions
+
+    function createAuction(
+        uint256 _nftId,
+        uint256 _minimumBid,
+        uint256 _auctionEnd
+    ) external nonReentrant ensureNFTOwner(_nftId) {
+        require(msg.sender != address(0), "Invalid caller address");
+        require(_auctionEnd > block.timestamp, "Invalid auctionEnd time");
+
+        uint256 auctionId = _auctionCounter.current();
+
+        Auction memory _auction = Auction({
+            id: auctionId,
+            nftContract: metawoodNFT,
+            nftId: _nftId,
+            creator: payable(msg.sender),
+            bidIncreaseThreshold: bidIncreaseThreshold,
+            auctionEnd: _auctionEnd,
+            minimumBid: _minimumBid,
+            highestBid: 0,
+            highestBidder: payable(address(0)),
+            bids: new uint256[](0),
+            bidders: new address[](0),
+            status: AuctionState.ACTIVE
+        });
+
+        metawoodNFT.safeTransferFrom(msg.sender, address(this), _nftId, 1, "0x00");
+
+        require(metawoodNFT.balanceOf(address(this), _nftId) == 1, "Nft not escrowed!");
+
+        _auctions[auctionId] = _auction;
+        _auctionCounter.increment();
+
+        emit AuctionCreated(
+            auctionId,
+            address(metawoodNFT),
+            _nftId,
+            msg.sender,
+            bidIncreaseThreshold,
+            _auctionEnd,
+            _minimumBid
+        );
+    }
+
+    function makeBid(uint256 _auctionId)
+        external
+        payable
+        nonReentrant
+        ensureValidAuction(_auctionId)
+        ensureActiveAuction(_auctionId)
+    {
+        Auction storage _auction = _auctions[_auctionId];
+        require(msg.sender != _auction.creator, "Cannont bid on own auction!");
+        require(block.timestamp <= _auction.auctionEnd, "Auction Deadline passed!");
+        require(msg.value >= _auction.minimumBid, "Bid amount greater than minimum bid required! ");
+
+        if (_auction.highestBid > 0) {
+            require(msg.value >= getNextBid(_auctionId), "Bid amount less than max bid threshold");
+        }
+
+        //return amount to old bidder
+        (bool success, ) = _auction.highestBidder.call{value:_auction.highestBid}("");
+        require(success, "Old highest bid payback failed");
```

Comment:
Lets use safeTransferETH from TransferHelper library I had added.
Will make all eth transfers safe and easy to write.
Please update this at all places

Reply:
Yup, was thinking of replacing them after the send-to-call change but it then skipped my mind.